### PR TITLE
Readme - user guide

### PR DESCRIPTION
**Pull request by gael-connan-cybex**
added a more detailed user guide

see https://github.com/cybex-gmbh/laravel-protector/tree/doc/provide-howto#readme for a formatted view

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,32 +2,110 @@
...
-Encrypt your database dumps with Sodium Encryption.
+Allows a client to download and import a live server database backup.
+
+Common usage scenarios
+- Non-productive developer machines download the live server database. 
+- A central backup server collects backups from multiple live servers.
+
+Feature set:
+ 
+- Download and Import databases from a server.
```
Comment (gael-connan-cybex) [inline README.md:13]: import

File: README.md [hunk 2]
```diff
@@ -2,32 +2,110 @@
...
+- the public key to give to your server admin
+- the private key to save in your .env file
+
+Your server admin will then give you the token and server-url to save in your .env file.
+Unless specified otherwise in your software, the .env keys are
+
+```bash
+PROTECTOR_AUTH_TOKEN=
+PROTECTOR_SERVER_URL=
+```
```
Comment (mszulik) [inline README.md:56]: Might want to add `PROTECTOR_PRIVATE_KEY` for the private key.

Comment (mszulik) [reply README.md:56]: Not too necessary tho since it is already stated when using the `protector:keys` command.

Comment (gael-connan-cybex) [reply README.md:56]: yes, this is currently regarded as a second, separate step

File: README.md (hunk 1 above)
Comment (gael-connan-cybex) [reply README.md]: fixed

**Review by mszulik — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -2,32 +2,110 @@
 
 [![Latest Version on Packagist](https://img.shields.io/packagist/v/cybex/laravel-protector.svg?style=flat-square)](https://packagist.org/packages/cybex/laravel-protector)
 
-Protect Databases by generating Backups and Import these on non-productive Environments.
-Authenticate users with the usage of Laravel Sanctum Tokens.
-Encrypt your database dumps with Sodium Encryption.
+Allows a client to download and import a live server database backup.
```
Comment (jheusinger) [inline README.md:5]: Description is lacking a lot of available features.

* Encrypted or plain backups
* Generation of local backups
* Import of existing backups
* Downloading backups from a remote source with or without import

File: README.md [hunk 4]
```diff
@@ -2,34 +2,196 @@
 
 [![Latest Version on Packagist](https://img.shields.io/packagist/v/cybex/laravel-protector.svg?style=flat-square)](https://packagist.org/packages/cybex/laravel-protector)
 
-Protect Databases by generating Backups and Import these on non-productive Environments.
-Authenticate users with the usage of Laravel Sanctum Tokens.
-Encrypt your database dumps with Sodium Encryption.
+Allows downloading, importing and exporting database backups.
```
Comment (lupinitylabs) [inline README.md:5]:
```suggestion
This package allows you to download, export and import your application's database backups.
```

File: README.md [hunk 5]
```diff
@@ -2,34 +2,196 @@
...
+>By default dumps are stored in `storage/app/protector`
+
+### Export to file
+
+To save a copy of your local database to your local disk, run
+```bash
+php artisan protector:export
+```
+
+## Setup Instructions
```
Comment (lupinitylabs) [inline README.md:65]:
```suggestion
## Setup instructions
```
Stick to one style in headings.

File: README.md [hunk 6]
```diff
@@ -2,34 +2,196 @@
...
 
 ```bash
 composer require cybex/laravel-protector
 ```
 
-## Usage
+You can optionally publish the protector config to set the following options
+- fileName: the file name of the database dump
+- baseDirectory: where files are being stored
+- diskName: a dedicated Laravel disk defined in config/filesystems.php. These can point to a specific folder, disk or a  cloud file bucket like AWS S3.
```
Comment (lupinitylabs) [inline README.md:84]:
```suggestion
- diskName: a dedicated Laravel disk defined in config/filesystems.php. These can point to a specific local folder or a cloud file bucket like AWS S3.
```
Laravel's filesystem disks are always folders, right? How would you set a disk?

File: README.md [hunk 7]
```diff
@@ -2,34 +2,196 @@
...
+```bash
+php artisan vendor:publish --tag=protector.migrations
+```
+
+Run the migration on the client and server repository.
+```bash
+php artisan migrate
+```
+
+#### On the Client
```
Comment (lupinitylabs) [inline README.md:148]:
```suggestion
#### On the client machine
```

File: README.md [hunk 8]
```diff
@@ -2,34 +2,196 @@
...
+PROTECTOR_SERVER_URL=
+```
+
+>Do not give your private key to anyone and keep it protected at all time
+
+See [Usage](#usage) on how to import the remote database.
+
+>Downloaded database dump files are stored unencrypted
+
+#### On the Server
```
Comment (lupinitylabs) [inline README.md:170]:
```suggestion
#### On the server
```

**Review by lupinitylabs — CHANGES_REQUESTED** [review]

File: README.md [hunk 9]
```diff
@@ -2,34 +2,196 @@
...
+You will receive the token and url to give back to the developer, who has to save them in their .env file.
+
+The developer can then download and import the server database on their own.
+
+### Setup for collecting backups from multiple servers
+
+You can develop a custom client that can access and store remote server backups. The servers can be different Laravel projects that have the protector package installed.
+
+See the previous chapter on how to give your backup client access to all servers. The backup client will need an according user on each target server.
+* All the backup users on the target servers will have the same public key from the client
```
Comment (lupinitylabs) [inline README.md:189]:
```suggestion
- All the backup users on the target servers will have the same public key from the client
```
It does not make a difference, but you used dashes everywhere else, so it seems odd to use asterisks here.

File: README.md [hunk 10]
```diff
@@ -2,34 +2,196 @@
...
+* For each target server, the client will store the according url and token
```
Comment (lupinitylabs) [inline README.md:190]:
```suggestion
- For each target server, the client will store the according url and token 
```

File: README.md [hunk 11]
```diff
@@ -2,34 +2,196 @@
...
+Your server admin will then give you the token and server-url to save in your .env file.
+Unless specified otherwise in your software, the .env keys are
+```bash
+PROTECTOR_AUTH_TOKEN=
+PROTECTOR_SERVER_URL=
+```
+
+>Do not give your private key to anyone and keep it protected at all time
+
+See [Usage](#usage) on how to import the remote database.
```
Comment (lupinitylabs) [inline README.md:166]:
```suggestion
See [Usage](#Usage) on how to import the remote database.
```
Anchor link does not seem to work within Github. Seems like it is case sensitive ( see https://github.com/fefong/markdown_readme )

Comment (gael-connan-cybex) [reply README.md:166]: https://github.com/cybex-gmbh/laravel-protector/tree/doc/provide-howto#readme it works, try it out here
github adds an anchor to headings and makes it kebap case

**Review by gael-connan-cybex — COMMENTED** [review]
didnt actually want to start a review

File: README.md [hunk 12]
```diff
@@ -2,34 +2,196 @@
...
+- A central backup server can collect backups from multiple live servers
+
+
+## Feature set
+
+- Download and optionally import databases from a server.
+- Import existing database files.
+- Export the local database to a file.
+- User authentication through Laravel Sanctum tokens.
+- Transport encryption using Sodium.
```
Comment (mszulik) [inline README.md:20]:
```suggestion
- Download and optionally import databases from a server
- Import existing database files
- Export the local database to a file
- User authentication through Laravel Sanctum tokens
- Transport encryption using Sodium
```

File: README.md (hunk 11 above)
Comment (gael-connan-cybex) [reply README.md:166]: taking my like and fleeing to hawaii

File: README.md (hunk 12 above)
Comment (gael-connan-cybex) [reply README.md]: fixed that seperately

**Pull request merged**